### PR TITLE
Fixed issue #334 Changed hard coded values according to TYPE_CODE

### PR DESCRIPTION
**Pull request by GovindaSharma**
Fixed issue #334 Changed hard coded values according to TYPE_CODE

### Description (*)
Fixed issue #334 Changed hard coded values according to TYPE_CODE

### Fixed Issues (if relevant)

1. magento/magento2 #334: Changed hard coded values according to TYPE_CODE
2. ...

### Manual testing scenarios (*)

   1. Changed hard coded 'Downloadable' to Magento\Downloadable\Model\Product\Type::TYPE_DOWNLOADABLE in
    app/code/Magento/DownloadableGraphQl/Model/DownloadableProductTypeResolver.php

  2.  Changed hard coded 'Bundle' to Magento\Bundle\Model\Product\Type::TYPE_CODE in
    app/code/Magento/BundleGraphQl/Model/BundleProductTypeResolver.php

  3.  Changed hard coded 'Bundle' to Magento\GroupedProduct\Model\Product\Type\Grouped::TYPE_CODE in
    app/code/Magento/GroupedProductGraphQl/Model/GroupedProductTypeResolver.php

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/graphql-ce?pullRequest=340) <br/>All committers have signed the CLA.

File: app/code/Magento/BundleGraphQl/Model/BundleProductTypeResolver.php [hunk 1]
```diff
@@ -19,7 +20,7 @@ class BundleProductTypeResolver implements TypeResolverInterface
      */
     public function resolveType(array $data) : string
     {
-        if (isset($data['type_id']) && $data['type_id'] == 'bundle') {
+        if (isset($data['type_id']) && $data['type_id'] == Type::TYPE_CODE) {
             return 'BundleProduct';
```
Comment (nuzil) [inline app/code/Magento/BundleGraphQl/Model/BundleProductTypeResolver.php]: Can you please move also this return value to constant?

File: app/code/Magento/DownloadableGraphQl/Model/DownloadableProductTypeResolver.php [hunk 2]
```diff
@@ -19,7 +20,7 @@ class DownloadableProductTypeResolver implements TypeResolverInterface
      */
     public function resolveType(array $data) : string
     {
-        if (isset($data['type_id']) && $data['type_id'] == 'downloadable') {
+        if (isset($data['type_id']) && $data['type_id'] == Type::TYPE_DOWNLOADABLE) {
             return 'DownloadableProduct';
```
Comment (nuzil) [inline app/code/Magento/DownloadableGraphQl/Model/DownloadableProductTypeResolver.php]: Can you please move also this return value to constant?

File: app/code/Magento/GroupedProductGraphQl/Model/GroupedProductTypeResolver.php [hunk 3]
```diff
@@ -19,7 +19,7 @@ class GroupedProductTypeResolver implements TypeResolverInterface
      */
     public function resolveType(array $data) : string
     {
-        if (isset($data['type_id']) && $data['type_id'] == 'grouped') {
+        if (isset($data['type_id']) && $data['type_id'] == Type::TYPE_CODE) {
             return 'GroupedProduct';
```
Comment (nuzil) [inline app/code/Magento/GroupedProductGraphQl/Model/GroupedProductTypeResolver.php]: Can you please move also this return value to constant?

**Review by nuzil — CHANGES_REQUESTED** [review]

**Comment by GovindaSharma** [comment]
hi @nuzil thanks for the update. so should i introduce new constant like COSNT GROUPED_PRODUCT = "GoupedProduct"?

**Comment by GovindaSharma** [comment]
Hi @nuzil i have updated the code as per your request.Please check it
Thank you

**Comment by GovindaSharma** [comment]
@nuzil any update on this?

**Comment by GovindaSharma** [comment]
@nuzil @naydav when this pull request is going to merge?

**Comment by naydav** [comment]
@GovindaSharma 
Looks like this code already merged
https://github.com/magento/magento2/blame/2.3-develop/app/code/Magento/BundleGraphQl/Model/BundleProductTypeResolver.php

**Comment by naydav** [comment]
The functionality has already been merged.

**Pull request closed without merging**